### PR TITLE
Gradle 7 support

### PR DESCRIPTION
**Pull request by sandorfehervari**
Gradle 7 support
 - added @Internal annotations for properties that are used internally in the plugin
 - Source directories are created using `ObjectFactory`
 - groovy 3 in tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,10 +31,10 @@ allprojects {
 }
 
 subprojects { subproject ->
-	apply plugin: 'maven'
 	apply plugin: 'maven-publish'
 	// For some reason this is needed ot have dependencies in POM files
 	apply plugin: "java"
```
Comment (lptr) [inline build.gradle:36]: You shouldn't need `java` if you also use `java-library`.

File: gradle-spaghetti-typescript-plugin/src/main/java/com/prezi/spaghetti/typescript/gradle/internal/MergeDtsTask.java [hunk 2]
```diff
@@ -31,7 +26,7 @@
     private File sourceDir;
     private File tsCompilerPath = null;
 
-    @Input
+    @Internal
     public File getWorkDir() {
         return workDir;
     }
```
Comment (lptr) [inline gradle-spaghetti-typescript-plugin/src/main/java/com/prezi/spaghetti/typescript/gradle/internal/MergeDtsTask.java:32]: I wonder why we never used `Task.getTemporaryDir()` for this instead. I guess we didn't know it existed. :) In any case, I guess it's good to have it now. We'll probably have something like `@TemporaryDirectory` in Gradle to mark these properties in the future.

**Review by lptr — COMMENTED** [review]
Nice to see Spaghetti living on! :) I was the original author, have since moved to work at Gradle itself. I left a few comments, hope you don't mind.

A couple more things that you could consider for future proofing Gradle plugins:

- migrate from `TaskContainer.create()` to the lazy `TaskContainer.register()`
- migrate from raw properties (`String getThing()` / `void setThing(String thing)`) to lazy properties (`Property<String> getThing()` etc.)

I'm planning to write up a blog post on the plans here once I find the time...

**Comment by sandorfehervari** [comment]
Thank you @lptr for the feedbacks, I will create cards about these possible improvements you've mentioned. I already started to refactor a bit to use properties in the gradle-haxe-plugin in a smaller part. At the moment I was focusing on the necessary changes in order to make it work with Gradle 7.

**Pull request merged**